hedyorg/hedy
# [BUG FIX] Fixes copy modal hiding when alert timeout was due

**Description**
In this PR we fix an issue where the hide and timeout functionality of the alerts was not working as expected. Because the timeout was not reset at all modals a situation could occur where the old timeout of the `copy modal` would make the newly shown modal disappear. This caused the bug as described in #1897.

**Fixes #1897**

**How to test**
Duplicate the behavior as described in #1897 and notice that this bug no longer occurs. Verify that the different modals are working in other places. A few simply ways to test the different modals are:

- Remove a program
- Re-name a class
- Save a program with a unique name
- Save a program with a duplicate name
- Try to repair a program

**Checklist**
Done? Check if you have it all in place using this list*
  

* If you're unsure about any of these, don't hesitate to ask. We're here to help!
